### PR TITLE
Adding S3 support to the ROSA pruner

### PR DESCRIPTION
**Pull request by bradmwilliams**

### CODE REVIEW

File: hack/rosa-cleanup.py [hunk 1]
```diff
@@ -82,49 +76,24 @@ def prune_load_balancers(clusters, resources):
...
-        for tag in lb['Tags']:
-            found = False
-            if tag['Key'] == 'api.openshift.com/id':
-                for cluster in clusters:
-                    if tag["Value"] == cluster['id']:
-                        found = True
-                        break
-                if not found:
-                    arns.append(lb["ResourceArn"])
+        if not associated_with_active_cluster(lb['Tags'], clusters):
```
Comment (DennisPeriquet) [inline hack/rosa-cleanup.py:79]: The logic here is the not the same as what `associated_with_active_cluster` replaces. But if the new logic in `associated_with_active_cluster` covers what you want, this is ok.

Comment (bradmwilliams) [reply hack/rosa-cleanup.py:79]: AFAICT, other than checking for a couple different tags at once, the logic should be pretty much identical

File: hack/rosa-cleanup.py [hunk 2]
```diff
@@ -198,16 +159,8 @@ def prune_ec2_snapshots(clusters, resources):
...
-            found = False
-            if tag['Key'] == 'api.openshift.com/id':
-                for cluster in clusters:
-                    if tag['Value'] == cluster['id']:
-                        found = True
-                        break
-                if not found:
-                    snapshots.append(ec2.Snapshot(snapshot["SnapshotId"]))
-                    break
+        if not associated_with_active_cluster(snapshot['Tags'], clusters):
```
[the remark this answers is not in the document]
Comment (bradmwilliams) [reply hack/rosa-cleanup.py:162]: AFAICT, other than checking for a couple different tags at once, the logic should be pretty much identical

**Comment by DennisPeriquet** [comment]
/hold
/lgtm

cancel the hold when ready to merge.

**Pull request merged**